### ISSUE TITLE
CA certificate for the data provider

### ISSUE THREAD

**Issue by stephan2012** [drakkan/sftpgo#682]
What is the correct way to pass a CA certificate for a data provider? To enable `full-verify` mode for Postgres connections, `sftpgo` somehow needs to know about it … There is no documented `ca_certificates` option.

**Comment by drakkan** [comment]
Hi,

you can configure a `connection_string`. Take a look at the CockroachDB configuration [here](https://github.com/drakkan/sftpgo/blob/main/docs/howto/getting-started.md#use-cockroachdb-data-provider). The postgres configuration should be quite similar

**Comment by stephan2012** [comment]
Thanks, @drakkan. Indeed, that could be a way. My only problem is that I can no longer pickup the username and password for the connection from a Kubernetes Secret (generated by the database operator) using `secretKeyRef` with this approach.

**Comment by stephan2012** [comment]
Ah, wait, something like this should work:

```yaml
  - name: SFTPGO_DATA_PROVIDER__USERNAME
    valueFrom:
      secretKeyRef:
        name: sftpgo.sftpgo-db.credentials.postgresql.acid.zalan.do
        key: username
  - name: SFTPGO_DATA_PROVIDER__PASSWORD
    valueFrom:
      secretKeyRef:
        name: sftpgo.sftpgo-db.credentials.postgresql.acid.zalan.do
        key: password
  - name: SFTPGO_DATA_PROVIDER__CONNECTION_STRING
    value: "postgresql://$(SFTPGO_DATA_PROVIDER__USERNAME):$(SFTPGO_DATA_PROVIDER__USERNAME)@$(SFTPGO_DATA_PROVIDER__HOST):$(SFTPGO_DATA_PROVIDER__PORT)/$(SFTPGO_DATA_PROVIDER__NAME)?&sslmode=verify-full&sslrootcert=%2Frun%2Fcerts%2Fdb%2Fca.crt&connect_timeout=10"
```

Anyway, it would be nice to have discrete config options for database TLS handling. :-)

**Comment by drakkan** [comment]
Hi,

I'm a bit busy in these days, I will evaluate this request better in the future, thanks for your suggestion

**Comment by stephan2012** [comment]
Thanks for your feedback. I’ll leave it up to you to keep this issue as a reminder or close it.

**Comment by drakkan** [comment]
a feedback would be appreciated, I haven't tested yet the client cert authentication and the mysql config

**Comment by stephan2012** [comment]
Thank you so much, @drakkan! I will try my best to test in the upcoming week.

**Comment by drakkan** [comment]
I did some basic tests, with postgres all the combinations seems to work. For mysql we can hit an edge case if someone set certificates and ssl mode to preferred. This is unlikely to happen let see if someone complains. 

However it seems that until now no one has used certificate authentication with mysql as the driver requires registration of a custom tls configuration and I only added it in the last commit.

An additional confirmation would be appreciated. You will find these new settings in v2.3.0
